src/overlays/ovl_i4/fox_ka.c

```diff
@@ -1,63 +1,2183 @@
+    f32 yDist;
+    f32 xRand;
+    f32 yRand;
+    f32 zRand;
+    f32 xAngle;
+    f32 yAngle;
+    f32 zAngle;
+    f32 xPos;
+    f32 yPos;
+    f32 zPos;
+    f32 xSin;
+    f32 ySin;
+    f32 xCos;
+    f32 yCos;
+    UnkEntity28* entity;
+    s32 pad2[9];
+
+    zPos = 0.0f;
+    yPos = 0.0f;
+    xPos = 0.0f;
+    Math_SmoothStepToF(&actor->fwork[10], 0.0f, 0.1f, 0.2f, 0.1f);
+    Math_SmoothStepToF(&actor->fwork[9], actor->fwork[10], 0.1f, 2.0f, 0.1f);
+    state = 0;
+    actor->iwork[5] = 0;
+    switch (actor->state) {
+        case 1:
+            actor->fwork[1] = 40.0f;
+            if (actor->timer_0BC == 0) {
+                actor->state = 3;
+            }
+            break;
+
+        case 2:
+            state = 1;
+            xDist = fabsf(actor->fwork[4] - actor->obj.pos.x);
+            yDist = fabsf(actor->fwork[6] - actor->obj.pos.z);
+            if (actor->unk_0E6 < 0) {
+                actor->state = 3;
+            } else {
+                if (actor->unk_0E4 >= 10) {
+                    xPos = __sinf(((actor->index * 45) + gGameFrameCount) * M_DTOR) * 200.0f;
+                    yPos = __cosf(((actor->index * 45) + (gGameFrameCount * 2)) * M_DTOR) * 200.0f;
+                    zPos = __sinf(((actor->index * 45) + gGameFrameCount) * M_DTOR) * 200.0f;
+                }
+                actor->fwork[4] = gActors[actor->unk_0E6].obj.pos.x + xPos;
+                actor->fwork[5] = gActors[actor->unk_0E6].obj.pos.y + yPos;
+                actor->fwork[6] = gActors[actor->unk_0E6].obj.pos.z + zPos;
+                actor->fwork[1] = gActors[actor->unk_0E6].fwork[0] + 10.0f;
+                if (actor->fwork[1] < 30.0f) {
+                    actor->fwork[1] = 30.0f;
+                }
+                actor->fwork[3] = 1.4f;
+                if (actor->unk_0E6 >= 0) {
+                    if (yDist < 800.0f) {
+                        if (xDist < 800.0f) {
+                            actor->fwork[1] = gActors[actor->unk_0E6].fwork[0] - 5.0f;
+                        }
+                    } else if (actor->timer_0C0 == 0) {
+                        actor->timer_0C0 = RAND_INT(200.0f) + 200;
+                        actor->fwork[10] = 20.0f;
+                    }
+                    if ((yDist < 1500.0f) && (xDist < 1500.0f)) {
+                        actor->iwork[4] += 1;
+                        actor->iwork[5] = 1;
+
+                        if (!((actor->index + gGameFrameCount) & 11) && (Rand_ZeroOne() < 0.1f) &&
+                            (func_80031900(actor) != 0) && (gActors->state == 2)) {
+                            actor->iwork[0] = 1;
+                        }
+                    } else {
+                        actor->iwork[4] = 0;
+                    }
+
+                    if ((actor->unk_0E6 > 0) &&
+                        ((gActors[actor->unk_0E6].obj.status == 3) || gActors[actor->unk_0E6].state == 6 ||
+                         gActors[actor->unk_0E6].obj.status == 0)) {
```

Comment:
```suggestion
                        ((gActors[actor->unk_0E6].obj.status == 3) || (gActors[actor->unk_0E6].state == 6) ||
                         (gActors[actor->unk_0E6].obj.status == 0))) {
```
also actor statuses have an enum